### PR TITLE
New review code branch

### PR DESCRIPTION
**Pull request by rong2**
Fast forwarded everything.

### CODE REVIEW

File: client/src/actions/smells.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+}
+
+export function fetchCodeSmells() {
+	return function (dispatch) {
+		dispatch(requestCodeSmells());
+		return request
+			.get('http://localhost:8000/app/codesmells/')
+			.end(function(err, res) {
+				if(err || !res.ok) {
```
Comment (rogerbutt) [inline client/src/actions/smells.js:28]: Is this the fix for error checking?

File: client/src/actions/smells.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                if(err || !res.ok) {
+                    console.log("add codesmells failure...");
+                } else {
+                    console.log('add codesmells success!');
+                }
+            });
+    }   
+} 
+
+export function addCodeSmells(user, codeid, line, codesmellname) {
```
Comment (rogerbutt) [inline client/src/actions/smells.js]: There are two functions named addCodeSmells?

File: client/src/reducers/code.js [hunk 3]
```diff
@@ -17,7 +26,26 @@ export default function codes(state = {}, action) {
...
+            let selectedLines = state.selectedLines || [];
+            let selectedLineObject = {};
+            selectedLineObject["line"] = action.payload.lineNumber;
+            selectedLineObject["smell"] = action.payload.codeSmellName;
+            selectedLines.push(selectedLineObject);
+            return Object.assign({}, state, {
+                selectedLines: _.uniqWith(selectedLines, _.isEqual)
+            });
+
+        case SUBMIT_CODE:
```
Comment (rogerbutt) [inline client/src/reducers/code.js:40]: There's a send code and submit code?

File: client/src/reducers/user.js [hunk 4]
```diff
@@ -18,18 +19,22 @@ export default function user(state = initialState, action) {
...
                 user: action.payload.user,
-                code: action.payload.code,
+                code: action.payload.code
             });
         case GET_USER_FAILURE:
             return Object.assign({}, state, {
-                isRequesting: false,
+                isRequesting: false
+            });
+        case SEND_USER:
```
Comment (rogerbutt) [inline client/src/reducers/user.js]: I don't really understand this SEND_USER. Can't you get the user id from the state.user.user.id?

File: client/src/actions/smells.js (hunk 1 above)
Comment (rong2) [inline client/src/actions/smells.js:28]: Yes, this is how superagent examples check for errors.

File: client/src/actions/smells.js (hunk 2 above)
Comment (rong2) [inline client/src/actions/smells.js]: Oops. That was my fault when I was rebasing.

File: client/src/reducers/code.js (hunk 3 above)
Comment (rong2) [inline client/src/reducers/code.js:40]: sendCode sends just the code object from SubmitCode to SubmitCodeSmells, which then takes the code, user and selectedLines and calls submitCode

File: client/src/reducers/user.js (hunk 4 above)
Comment (rong2) [inline client/src/reducers/user.js]: Nope, it's undefined unless I explicitly send it.

Alternatively, I can remove SEND_USER and just call getUserInfo() again in submitCodeSmells

File: client/src/reducers/code.js (hunk 3 above)
Comment (rogerbutt) [inline client/src/reducers/code.js:40]: Maybe just make the names a little more different so there some differentiation

**Pull request merged**